src/test/java/com/serotonin/bacnet4j/obj/ObjectTestUtils.java

```diff
@@ -0,0 +1,147 @@
+package com.serotonin.bacnet4j.obj;
+
+import static org.junit.Assert.fail;
+
+import java.time.Clock;
+import java.util.LinkedList;
+import java.util.List;
+
+import com.serotonin.bacnet4j.type.Encodable;
+import com.serotonin.bacnet4j.type.enumerated.PropertyIdentifier;
+
+public class ObjectTestUtils {
+    /**
+     * Test utility to wait for a specific value to be written to an object.
+     */
+    public static class ObjectWriteNotifier<T extends BACnetObject> extends AbstractMixin {
+        private final T bo;
+        private final boolean debug;
+        private final List<WrittenProperty> writtenProperties = new LinkedList<>();
+
+        /**
+         * You can call this directly, but you probably want to use createObjectWriteNotifier below.
+         *
+         * @param bo the object to which to listen
+         * @param debug whether debugging messages should be written to stdout. This is done instead of logging because
+         *              logging requires extra work to enable messages to be seen, which also includes the logging of a
+         *              lot of other messages.
+         */
+        public ObjectWriteNotifier(T bo, boolean debug) {
+            super(bo);
+            this.bo = bo;
+            this.debug = debug;
+        }
+
+        /**
+         * This method allows the notifier to be the only object reference required. I.e. the client code doesn't need
+         * to maintain a reference to the object and the notifier. It can just ask the notifier for the object.
+         *
+         * @return the object which is being monitored
+         */
+        public T obj() {
+            return bo;
+        }
+
+        /**
+         * Useful for preventing the matching of initialization writes, or other writes that may be inconsequential to
+         * the test.
+         */
+        public void clear() {
+            synchronized (this) {
+                writtenProperties.clear();
+            }
+        }
+
+        @Override
+        protected void afterWriteProperty(PropertyIdentifier pid, Encodable oldValue, Encodable newValue) {
+            if (debug) {
+                System.out.println("DEBUG: PropertyWritten: "+ bo.getId() +", pid: "+ pid +", oldValue: "+ oldValue +", newValue: "+ newValue);
+            }
+            synchronized (this) {
+                // Write all written properties to the list of seen writes.
+                writtenProperties.add(new WrittenProperty(pid, newValue));
+                notifyAll();
+            }
```

Comment:
I suspect we could replace this synchronized/wait/notify logic with a `java.util.concurrent.SynchronousQueue`

Reply:
Hmm, not sure we want to block the core code?

Reply:
Yeah, that's simpler